tests/testthat/test-file.R

```diff
@@ -0,0 +1,135 @@
+
+context("file-based keyring")
+
+test_that("specify keyring explicitly", {
+
+  service_1 <- random_service()
+  username <- random_username()
+  password <- random_password()
+  keyring <- random_keyring()
+
+  kb <- backend_file$new(keyring = keyring)
+
+  expect_true(kb$keyring_is_locked())
+  expect_silent(kb$keyring_unlock(password = random_password()))
```

Comment:
So the first `unlock()` sets the password? Why not ask for it interactively like the other backends?

Reply:
`keyring_unlock()` will ask the for the password whenever it is not specified or passed as `NULL`. I think this is consistent with the other backends, no? Do you want me to change something about this behavior?

Reply:
I guess what I am asking is, in this is case what is the password of the keyring? That was not set in `$new()`.

Reply:
I'm sorry, but I'm not sure I understand what you are getting at.

The constructor does very little due to the way the api functions are set up:
- assume the api fun `key_get()` is called with non-default keyring
   ```r
   key_get <- function(service, username = NULL, keyring = NULL) {
     assert_that(is_non_empty_string(service))
     assert_that(is_string_or_null(username))
     default_backend()$get(service, username, keyring = keyring)
   }
   ```
- if the constructor wanted to set the keyring password, the user would be asked to enter two pwds: first the one for the default keyring and then the one for the keyring he/she wants to access
- the situation could be improved if the api funs would pass the keyring info to the constructor, something like
   ```r
   key_get <- function(service, username = NULL, keyring = NULL) {
     assert_that(is_non_empty_string(service))
     assert_that(is_string_or_null(username))
     default_backend(keyring)$get(service, username)
   }
   ```
- I thought it might be a bit presumptuous of me to ask for such a change, so I decided on working around this, by only asking for the keyring pwd on unlock
- as it turns out, it might not even make sense to try and set the keyring pwd in the constructor, as functions such as `has_keyring_support()` or `keyring_is_locked()` would then also ask for pwds which they don't need

Am I making sense?

Reply:
If the user does not unlock the keyring by him/herself, like so,

```r
kb <- backend_file$new(keyring = random_keyring())
kb$set_with_value(random_service(), random_username(), random_password())
```

`keyring_unlock()` is called by `set_with_value()`, in which case, the user will be prompted to enter the keyring password.

Reply:
Makes sense!